jnosql-dynamodb/src/main/java/org/eclipse/jnosql/databases/dynamodb/communication/DefaultDynamoDBDocumentManager.java

```diff
@@ -0,0 +1,348 @@
+        requireNonNull(ttl, "ttl is required");
+        documentEntity.add(getTTLAttributeName(documentEntity.name()).get(), Instant.now().plus(ttl).truncatedTo(ChronoUnit.SECONDS));
+        return insert(documentEntity);
+    }
+
+    @Override
+    public Iterable<DocumentEntity> insert(Iterable<DocumentEntity> entities) {
+        requireNonNull(entities, "entities are required");
+        return StreamSupport.stream(entities.spliterator(), false)
+                .map(this::insert)
+                .toList();
+    }
+
+    @Override
+    public Iterable<DocumentEntity> insert(Iterable<DocumentEntity> entities, Duration ttl) {
+        requireNonNull(entities, "entities is required");
+        requireNonNull(ttl, "ttl is required");
+        return StreamSupport.stream(entities.spliterator(), false)
+                .map(e -> this.insert(e, ttl))
+                .toList();
+    }
+
+    @Override
+    public DocumentEntity update(DocumentEntity documentEntity) {
+        requireNonNull(documentEntity, "entity is required");
+        Map<String, AttributeValue> itemKey = getItemKey(documentEntity);
+        Map<String, AttributeValueUpdate> attributeUpdates = asItemToUpdate(documentEntity);
+        itemKey.keySet().forEach(attributeUpdates::remove);
+        dynamoDbClient().updateItem(UpdateItemRequest.builder()
+                .tableName(createTableIfNeeded(documentEntity.name()).table().tableName())
+                .key(itemKey)
+                .attributeUpdates(attributeUpdates)
+                .build());
+        return documentEntity;
+    }
+
+    private Map<String, AttributeValue> getItemKey(DocumentEntity documentEntity) {
+        DescribeTableResponse describeTableResponse = this.tables.computeIfAbsent(documentEntity.name(), this::getDescribeTableResponse);
+        Map<String, AttributeValue> itemKey = describeTableResponse
+                .table()
+                .keySchema()
+                .stream()
+                .map(attribute -> Map.of(attribute.attributeName(),
+                        toAttributeValue(documentEntity.find(attribute.attributeName(), Object.class).orElse(null))))
+                .reduce(new HashMap<>(), (a, b) -> {
+                    a.putAll(b);
+                    return a;
+                });
+        itemKey.put(getEntityAttributeName(), toAttributeValue(documentEntity.name()));
+        return itemKey;
+    }
+
+    private Map<String, AttributeValueUpdate> asItemToUpdate(DocumentEntity documentEntity) {
+        return toItemUpdate(this::resolveEntityNameAttributeName, documentEntity);
+    }
+
+    @Override
+    public Iterable<DocumentEntity> update(Iterable<DocumentEntity> entities) {
+        requireNonNull(entities, "entities is required");
+        return StreamSupport.stream(entities.spliterator(), false)
+                .map(this::update)
+                .toList();
+    }
+
+    @Override
+    public void delete(DocumentDeleteQuery documentDeleteQuery) {
```

Comment:
You can use the MongoDB driver as a sample. It was replaced to `DeleteQuery`

Reply:
Fixed!